src/data_provider/src/osinfo/sysOsParsers.cpp

```diff
@@ -392,6 +391,36 @@ bool MacOsParser::parseSwVersion(const std::string& in, nlohmann::json& output)
     return ret;
 }
 
+bool MacOsParser::parseSystemProfiler(const std::string& in, nlohmann::json& output)
+{
+    constexpr auto SEPARATOR{':'};
+    static const std::map<std::string, std::string> KEY_MAP
+    {
+        {"System Version", "os_name"},
+    };
+    std::stringstream data{in};
+    nlohmann::json tmp;
+    auto ret{ parseUnixFile(KEY_MAP, SEPARATOR, data, tmp) };
+
+    if (ret)
+    {
+        constexpr auto PATTERN_MATCH{R"(([^\s]+) ([^\s]+) ([^\s]+))"};
```

Comment:
Improve the regex by
- Only use extraction group to data of intrest
- Add start of / end of match